sdks/python/apache_beam/yaml/yaml_provider.py

```diff
@@ -730,36 +734,50 @@ def _path(cls, base_python, packages):
   def _create_venv_from_scratch(cls, base_python, packages):
     venv = cls._path(base_python, packages)
     if not os.path.exists(venv):
-      subprocess.run([base_python, '-m', 'venv', venv], check=True)
-      venv_python = os.path.join(venv, 'bin', 'python')
-      subprocess.run([venv_python, '-m', 'ensurepip'], check=True)
-      subprocess.run([venv_python, '-m', 'pip', 'install'] + packages,
-                     check=True)
-      with open(venv + '-requirements.txt', 'w') as fout:
-        fout.write('\n'.join(packages))
+      try:
+        subprocess.run([base_python, '-m', 'venv', venv], check=True)
+        venv_python = os.path.join(venv, 'bin', 'python')
+        subprocess.run([venv_python, '-m', 'ensurepip'], check=True)
+        subprocess.run([venv_python, '-m', 'pip', 'install'] + packages,
+                       check=True)
```

Comment:
I have a branch with most of these changes as well (just hadn't gotten around to cleaning it and opening a PR yet), but one change I noticed I had that isn't here is for the `pip install` commands. For some reason subprocess does not cause the execution to fail when the command does not pass (despite `check=True`), but I noticed that `python -m pip install ...` was not actually running, so dependencies were not getting installed. This change worked for me:
```suggestion
        venv_python = os.path.join(venv, 'bin', 'python')
        venv_pip = os.path.join(venv, 'bin', 'pip')
        subprocess.run([venv_python, '-m', 'ensurepip'], check=True)
        subprocess.run([venv_pip, 'install'] + packages,
                       check=True)
```